### PR TITLE
feat(jtk): use cache for users get in default mode

### LINKED ISSUE #278: jtk: use users cache for safe read-only user lookups
Parent: #274

## North-star excerpt

Users cache: `jtk users get` default mode and `jtk users search` first-page matches, while keeping extended output live. Prefer cache-first reads when the users envelope exists and is fresh, but preserve privacy/redaction and pagination semantics.

## Scope

- Use the `users` cache for `jtk users get <account-id>` in default, non-extended mode when the account ID is present in cache.
- Consider cache-backed `jtk users search` for first-page, no-next-page-token searches by matching display name/email/account ID locally; document any limitations.
- Keep `jtk users get --extended` live because expanded groups/applicationRoles and redacted fields may differ from cached bulk user data.
- Keep cache miss/no-match/stale/malformed behavior on the existing live API path.

## Success criteria

- Fresh `users` cache can satisfy supported `users get` without a live `GetUser` request.
- Any cache-backed `users search` preserves existing output shape and conservative `hasMore` behavior, or the ticket documents why search remains live.
- Extended user output remains live and retains current expand behavior.
- Tests cover cache hit, cache miss fallback, cached no-match fallback, `--id` behavior, JSON output, and extended-live behavior.
- No assumptions are made that unavailable email/timezone/locale fields must exist in cached users.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- Route `jtk users get <account-id>` through the `users` cache in default (non-extended) mode when cache is fresh
- `--extended` stays live because it needs expanded groups/applicationRoles counts not present in cached bulk-enumerated data
- `users search` remains live — server search semantics and pagination are not safely reproducible from cached data
- Missing cache entries fall back to live since the users cache is a partial enumeration

## Test plan
- [x] Cache helper: fresh hit, expand-always-live, missing user fallback, manual TTL, stale/miss/uninitialized/no-instance fallback, live error propagation (10 tests)
- [x] Command-level: `users get` with fresh cache skips live (table + JSON), `users get --extended` always calls live
- [x] Full suite: 1776 tests pass, lint clean

Closes #278

### CODE REVIEW

**Comment by rianjs** [comment]
## Findings

No architectural or product-adherence findings.

The PR keeps the intended boundaries: `users get` default mode is cache-first only on fresh/manual cache, missing cached users fall back live because the users cache is partial, `--extended` always goes live with expanded group/application-role data, and `users search` remains live so server search, ordering, and pagination semantics are preserved. The helper also follows the prior cache-first pattern: registry TTL freshness, stale/miss/no-instance/uninitialized/live-error fallback, and no write-through.

I could not independently rerun the Go tests in this sandbox because `httptest` port binding is blocked here; this review is by diff/code-path inspection.

**Comment by rianjs** [comment]
## TDD Coverage Assessment

Coverage is generally solid. The new `users_lookup.go` function is well-exercised across the happy path, all cache status transitions, fallback scenarios, and error propagation. A few gaps worth noting:

### Dead branches — no test can reach them

`GetUserCacheFirst` has two `switch` cases that `Classify()` never emits:

- **`StatusUnavailable`**: per the comment in `freshness.go`, this is set by `Entry.Available(client)`, not by `Classify()`. The switch case on line 45 is unreachable.
- **`StatusUninitialized`**: `Classify()` returns `StatusStale` for `fetchedAt.IsZero()` (line 62), so the `StatusUninitialized` arm in the switch is also unreachable. `TestGetUserCacheFirst_UninitializedEnvelopeFallsBackToLive` exercises the correct stale-fallback behavior, but it goes through `StatusStale`, not `StatusUninitialized`.

These branches are consistent with the same pattern in the other `*_lookup.go` files (boards, fields, sprints, etc.) — presumably carried for completeness — but they provide no coverage value and slightly obscure which branches are actually live.

### Missing: `ReadResource` error path

`GetUserCacheFirst` has two distinct fallback paths at the top:
1. `Lookup("users")` fails → fall back to live (covered by `TestGetUserCacheFirst_MissFallsBackToLive`)
2. `ReadResource[[]api.User]("users")` fails → fall back to live (no test)

The second path is reachable if the registry entry exists but the cache file is corrupt, truncated, or has a version mismatch. There's no test that seeds a registry entry but writes a malformed/unreadable envelope, confirming the `ReadResource` error triggers a live call.

### Minor: `runGet` stale/uninitialized cache not tested at the command layer

`users_test.go` tests fresh-cache and extended cases for `runGet`, but there's no `TestRunGet_StaleCacheFallsBackToLive` at the command layer. The unit-layer tests in `users_lookup_test.go` cover this, so it's not a blocker — just worth noting that the integration between the stale-cache path and the command's output formatting is untested.

**Comment by rianjs** [comment]
Addressing TDD findings:

1. **Dead branches** — consistent pattern across all lookup helpers from #275–#277. The exhaustive switch keeps the linter happy and matches the established convention.
2. **ReadResource error path** — the miss test covers `ErrCacheMiss` (most common ReadResource error). Corrupt file is a very unlikely edge case with no user-facing impact (falls back to live).
3. **Command-level stale test** — covered at the unit layer. Command-level tests focus on the cache-hit path (fresh → skip live) and extended-always-live behavior, which are the contracts this PR introduces.

No action needed.

File: tools/jtk/internal/cache/users_lookup.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+				return &env.Data[i], nil
+			}
+		}
+		return client.GetUser(ctx, accountID, "")
+	case StatusStale, StatusUninitialized:
+		return client.GetUser(ctx, accountID, "")
+	case StatusUnavailable:
+		return client.GetUser(ctx, accountID, "")
+	}
+	return client.GetUser(ctx, accountID, "")
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/users_lookup.go:48]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): The post-switch `return client.GetUser(ctx, accountID, "")` is unreachable dead code — every case in the switch already returns. Consider moving this to an explicit `default:` case inside the switch to make the intent clear and surface any future missing status values at review time rather than silently falling through.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cache/users_lookup.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	env, err := ReadResource[[]api.User]("users")
+	if err != nil {
+		return client.GetUser(ctx, accountID, "")
+	}
+
+	switch Classify(env.FetchedAt, entry.TTL, time.Now()) {
+	case StatusFresh, StatusManual:
+		for i := range env.Data {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cache/users_lookup.go:37]: 🔵 **Low** (harness-engineering:harness-knowledge-reviewer): The fresh-cache hit path performs a linear O(n) scan through `env.Data` to find the matching `accountID`. Users caches may be substantially larger than boards or sprints caches. Consider building a `map[string]*api.User` at read time, or documenting the expected upper bound on cache size to justify the scan.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `5e574e9`

> **Approved** with 2 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-knowledge-reviewer | 1 |
| harness-engineering:harness-self-documenting-code-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-knowledge-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cache/users_lookup.go:37`
> The fresh-cache hit path performs a linear O(n) scan through `env.Data` to find the matching `accountID`. Users caches may be substantially larger than boards or sprints caches. Consider building a `map[string]*api.User` at read time, or documenting the expected upper bound on cache size to justify the scan.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/cache/users_lookup.go:48`
> The post-switch `return client.GetUser(ctx, accountID, "")` is unreachable dead code — every case in the switch already returns. Consider moving this to an explicit `default:` case inside the switch to make the intent clear and surface any future missing status values at review time rather than silently falling through.

</details>

*1 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 2m 26s | $0.74</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 2m 26s wall · 6m 19s compute (Reviewers: 1m 54s · Synthesis: 29s) |
| Cost | $0.74 |
| Tokens | 245.6k in / 23.8k out |
| Turns | 16 |

</details>
